Delete torch::deploy from pytorch core

PaliC opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #85901

As we have migrated torch::deploy over to https://github.com/pytorch/multipy, we can now delete it from pytorch core as ongoing development will happen there.

This PR was created due to syncing issues with https://github.com/pytorch/pytorch/pull/85443 which is where the review history can be found.

Differential Revision: [D39933787](https://our.internmc.facebook.com/intern/diff/D39933787/)

**NOTE FOR REVIEWERS**: This PR has internal Meta-specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D39933787/)!

cc @ezyang @SherlockNoMad @soumith @EikanWang @jgong5 @wenzhe-nrv

PaliC committed:
Delete torch::deploy from pytorch core

As we have migrated torch::deploy over to https://github.com/pytorch/multipy, we can now delete it from pytorch core as ongoing development will happen there.

This PR was created due to syncing issues with https://github.com/pytorch/pytorch/pull/85443 which is where the review history can be found.

Differential Revision: [D39933787](https://our.internmc.facebook.com/intern/diff/D39933787/)

**NOTE FOR REVIEWERS**: This PR has internal Meta-specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D39933787/)!

[ghstack-poisoned]

PaliC committed:
Update on "Delete torch::deploy from pytorch core"


As we have migrated torch::deploy over to https://github.com/pytorch/multipy, we can now delete it from pytorch core as ongoing development will happen there.

This PR was created due to syncing issues with https://github.com/pytorch/pytorch/pull/85443 which is where the review history can be found.

Differential Revision: [D39933787](https://our.internmc.facebook.com/intern/diff/D39933787/)

**NOTE FOR REVIEWERS**: This PR has internal Meta-specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D39933787/)!

[ghstack-poisoned]

d4l3k reviewed (approved):
LGTM

d4l3k commented on torch/csrc/Module.cpp:
```diff
@@ -390,23 +390,18 @@ static PyObject* THPModule_parallelInfo(PyObject* module, PyObject* noargs) {
+    // the dlMTensor has been consumed
+    // PyCapsule_GetPointer has set an error indicator
+    PyErr_Clear();
+  }
```
why are we changing this code?

PaliC replied:
fixed was a rebasing issue

PaliC committed:
Update on "Delete torch::deploy from pytorch core"


As we have migrated torch::deploy over to https://github.com/pytorch/multipy, we can now delete it from pytorch core as ongoing development will happen there.

This PR was created due to syncing issues with https://github.com/pytorch/pytorch/pull/85443 which is where the review history can be found.

Differential Revision: [D39933787](https://our.internmc.facebook.com/intern/diff/D39933787/)

**NOTE FOR REVIEWERS**: This PR has internal Meta-specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D39933787/)!

[ghstack-poisoned]

PaliC committed:
Update on "Delete torch::deploy from pytorch core"


As we have migrated torch::deploy over to https://github.com/pytorch/multipy, we can now delete it from pytorch core as ongoing development will happen there.

This PR was created due to syncing issues with https://github.com/pytorch/pytorch/pull/85443 which is where the review history can be found.

Differential Revision: [D39933787](https://our.internmc.facebook.com/intern/diff/D39933787/)

**NOTE FOR REVIEWERS**: This PR has internal Meta-specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D39933787/)!

[ghstack-poisoned]

PaliC committed:
Update on "Delete torch::deploy from pytorch core"


As we have migrated torch::deploy over to https://github.com/pytorch/multipy, we can now delete it from pytorch core as ongoing development will happen there.

This PR was created due to syncing issues with https://github.com/pytorch/pytorch/pull/85443 which is where the review history can be found.

Differential Revision: [D39933787](https://our.internmc.facebook.com/intern/diff/D39933787/)

**NOTE FOR REVIEWERS**: This PR has internal Meta-specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D39933787/)!

[ghstack-poisoned]
